File: src/VisualStudio/CSharp/Impl/CodeModel/Extenders/CodeTypeLocationExtender.cs
// Licensed to the .NET Foundation under one or more agreements.
// The .NET Foundation licenses this file to you under the MIT license.
// See the LICENSE file in the project root for more information.

#nullable disable

using System.Runtime.InteropServices;
using Microsoft.VisualStudio.LanguageServices.CSharp.CodeModel.Interop;
using Microsoft.VisualStudio.LanguageServices.Implementation.Interop;

namespace Microsoft.VisualStudio.LanguageServices.CSharp.CodeModel.Extenders;

[ComVisible(true)]
[ComDefaultInterface(typeof(ICSCodeTypeLocation))]
public class CodeTypeLocationExtender : ICSCodeTypeLocation
{
    internal static ICSCodeTypeLocation Create(string externalLocation)
    {
        var result = new CodeTypeLocationExtender(externalLocation);
        return (ICSCodeTypeLocation)ComAggregate.CreateAggregatedObject(result);
    }

    private CodeTypeLocationExtender(string externalLocation)
        => ExternalLocation = externalLocation;

    public string ExternalLocation { get; }
}

File: src/VisualStudio/VisualBasic/Impl/CodeModel/Extenders/CodeTypeLocationExtender.vb
' Licensed to the .NET Foundation under one or more agreements.
' The .NET Foundation licenses this file to you under the MIT license.
' See the LICENSE file in the project root for more information.

Imports System.Runtime.InteropServices
Imports Microsoft.VisualStudio.LanguageServices.Implementation.Interop
Imports Microsoft.VisualStudio.LanguageServices.VisualBasic.CodeModel.Interop

Namespace Microsoft.VisualStudio.LanguageServices.VisualBasic.CodeModel.Extenders
    <ComVisible(True)>
    <ComDefaultInterface(GetType(IVBCodeTypeLocation))>
    Public Class CodeTypeLocationExtender
        Implements IVBCodeTypeLocation

        Friend Shared Function Create(externalLocation As String) As IVBCodeTypeLocation
            Dim result = New CodeTypeLocationExtender(externalLocation)
            Return CType(ComAggregate.CreateAggregatedObject(result), IVBCodeTypeLocation)
        End Function

        Private ReadOnly _externalLocation As String

        Private Sub New(externalLocation As String)
            _externalLocation = externalLocation
        End Sub

        Public ReadOnly Property ExternalLocation As String Implements IVBCodeTypeLocation.ExternalLocation
            Get
                Return _externalLocation
            End Get
        End Property
    End Class
End Namespace
